packages/office-ui-fabric-react/src/components/Button/Button.types.ts

```diff
@@ -290,6 +290,24 @@ export interface IButtonProps
   description?: IStyle;
 }
 
+export interface IButtonStyleProps {
+  /**
+   * Theme provided by High-Order Component.
+   */
+  theme?: ITheme;
+
+  /**
+   * Accept custom classNames
+   */
+  className?: string;
+
+  /**
+   * Changes the visual presentation of the button to be emphasized (if defined)
+   * @defaultvalue false
+   */
+  primary?: boolean;
+}
```

Comment:
Just discussing with Mak in person about making `IButtonStyleProps` simply `IButtonProps`. I don't think there's a good reason to exclude other button props and, since there are no 'style only' props to add, they could simply be the same interface.

Reply:
(Although now looking at the PR more it seems this will affect all of the getClassName functions that pass a more limited set of props.)